src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/converter/TimestampingServiceConverter.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * The MIT License
+ * Copyright (c) 2018 Estonian Information System Authority (RIA),
+ * Nordic Institute for Interoperability Solutions (NIIS), Population Register Centre (VRK)
+ * Copyright (c) 2015-2017 Estonian Information System Authority (RIA), Population Register Centre (VRK)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.niis.xroad.restapi.converter;
+
+import com.google.common.collect.Streams;
+import org.niis.xroad.restapi.openapi.model.TimestampingService;
+import org.niis.xroad.restapi.service.GlobalConfService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Converter for timestamping services related data between openapi and service domain classes
+ */
+@Component
+public class TimestampingServiceConverter {
+
+    private final GlobalConfService globalConfService;
```

Comment:
We have tried to avoid calling services from converters to keep them nicely decoupled and the codebase consistent. Maybe change the converter to use `GlobalConfFacade` and call the required facade methods directly (`globalConfFacade.getApprovedTspName(globalConfFacade.getInstanceIdentifier(), url);`)

Reply:
The implementation was updated based on your suggestion.